misc/pytorch_toolkit/breast_ultrasound_simulation/src/solve.py

```diff
@@ -0,0 +1,335 @@
+        self.adversarial_loss = self.adv_l(discriminator_output, self.y)
+
+        self.d_opt.zero_grad()
+        self.adversarial_loss.backward()
+        self.d_opt.step()
+
+    def train(self):
+
+        args = self.args
+        for epoch in range(args.epochs):
+            L1_epoch = 0
+            AdvG_epoch = 0
+            AdvD_epoch = 0
+            DisAcc_epoch = 0
+
+            print(
+                "\nEpoch = {}  Lr = {}".format(
+                    epoch,
+                    self.g_opt.state_dict()["param_groups"][0]["lr"]))
+            start_time = time.time()
+            for i_batch, sample in enumerate(self.train_data):
+                X, real, _ = sample
+
+                X = X.to(device)
+
+                self.real = real.to(device)
+
+                # generating fake images
+                self.fake = self.gen(X)
+
+                # channel shuffler with labels
+                self.discriminator_input, self.y = self.shuffler(
+                    self.real, self.fake)
+
+                self.set_requires_grad(self.dis, False)
+                self.g_back()
+                self.set_requires_grad(self.dis, True)
+
+                self.set_requires_grad(self.gen, False)
+                self.d_back()
+                self.set_requires_grad(self.gen, True)
+
+                # Accuracy of disc
+                temp = (
+                    self.discriminator_output.squeeze() > 0.5).type(
+                    torch.cuda.FloatTensor)
+                DisAcc_epoch += (self.y ==
+                                 temp).type(torch.cuda.FloatTensor).mean().item()
+
+                # Cumulative loss for analysis
+                L1_epoch += self.reconstruction_loss.item()
+                AdvG_epoch += self.generator_adversarial_loss.item()
+                AdvD_epoch += self.adversarial_loss.item()
+                # print(time.time())
+                # Python Logging
+                print('\rStep [{}/{}], L1: {:.4f}, AccD: {:.4f}, AdvD: {:.4f}, AdvG: {:.4f} Time: {:.1f}sec'.format((i_batch + 1),
+                                                                                                                    len(self.train_data),
+                                                                                                                    L1_epoch / (i_batch + 1),
+                                                                                                                    DisAcc_epoch / (i_batch + 1),
+                                                                                                                    AdvD_epoch / (i_batch + 1),
+                                                                                                                    AdvG_epoch / (i_batch + 1),
+                                                                                                                    time.time() - start_time),
+                      end=" ")
```

Comment:
Line is too long and could be displayed weird. Please, format this to have maximum line length 120 symbols.